mautic/mautic
# Mautic focus - engage consumers with bars, modals, notifications and/or full page alerts

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | todo
| Related developer documentation PR URL | na
| Issues addressed (#s or URLs) | 
| BC breaks? | n
| Deprecations? | n

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This brings mPower Focus, an exit intent feature previously only available to mautic.com SaaS subscribers to the community - for free!

The CorePermissions class had to be updated a bit to better support plugin permissions. All permissions and roles should function exactly the same - both developer and frontend use. It was updated to simply build a list of permission names and the classes associated when the cache is built. That list is then used to generate the permission class when needed rather than splicing together assumptions based on namespace, etc. 

#### Steps to test this PR:
1. If you're familiar with mPower Focus on Mautic.com, test away :-) It should function the same.
2. Otherwise, apply the PR, clear your cache, run migrations and you'll find Focus Items under Channels. Start playing :-) User documentation to come.
3. All permissions and roles should continue to work as is.
